### PR TITLE
Auth/[PM-53] - add rbac for organizations

### PR DESCRIPTION
**Pull request by dgoodman-bw**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Add RBAC for the organization entity.

## Code changes

* **OrganizationsController.cs** Applied action filter attribute to appropriate actions, and replaced the `model.ToOrganization` call with a new function so that permission checks can be applied to the value changes.
* **Permissions.cs** Added a new permission and renamed an existing one to be more accurate
* **OrganizationEditModel.cs** Removed the `ToOrganization` function because it was replaced with a function in the controller so permission checks could be applied.
* **RolePermissionMapping.cs** Added the new `Org_CheckEnabledBox` permission to the Owner, Admin, and Sales roles. Added the `Org_Plan_View` and `Org_Plan_Edit` permissions to the Sales role as this was previously missed.
* **Organizations/Edit.cshtml** Added permission checks to the UI elements, and replaced the Name and Business Name input text boxes with plain text as these fields can no longer be edited by any role.
* **_Layout.cshtml** Added permission check to the Organizations option in the nav bar.

**Owner role view**
Owner

**Admin role view**
Admin

**CS role view**
CS

**Billing role view**
Billing

**Sales role view**
Sales

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Admin/Controllers/OrganizationsController.cs
Comment (JaredSnider-Bitwarden) [inline src/Admin/Controllers/OrganizationsController.cs:1]: Clean!

File: src/Admin/Views/Organizations/Edit.cshtml [hunk 1]
```diff
@@ -117,228 +137,258 @@
...
-                    <div class="input-group-append">
-                        <button class="btn btn-secondary" type="button" id="gateway-customer-link">
-                            <i class="fa fa-external-link"></i>
-                        </button>
+        <div class="row">
+            <div class="col-sm">
+                <div class="form-group">
+                    <label asp-for="GatewayCustomerId"></label>
+                    <div class="input-group">
+                        <input type="text" class="form-control" asp-for="GatewayCustomerId" readonly='@(!canEditBilling)'>
```
Comment (JaredSnider-Bitwarden) [inline src/Admin/Views/Organizations/Edit.cshtml:343]: ❓ of curiosity: you use readonly for this input, but disabled for the gateway select on line 330. Is that intentional?

Comment (dgoodman-bw) [reply src/Admin/Views/Organizations/Edit.cshtml:343]: Great question. `readonly` is preferable in a lot of ways - the conditional syntax is more readable, and the values of `readonly` elements bind to the model on save, unlike `disabled` values. But unfortunately `readonly` can’t be applied to checkboxes or radio buttons, so `disabled` must be used instead.

Comment (JaredSnider-Bitwarden) [reply src/Admin/Views/Organizations/Edit.cshtml:343]: @dgoodman-bw, ah that makes sense. Thank you!

**Pull request merged**